### PR TITLE
Add star rating component

### PR DESCRIPTION
**Pull request by JStein92**

### CODE REVIEW

File: app/components/star-rating/app-star-rating.component.ts [hunk 1]
```diff
@@ -0,0 +1,56 @@
+import angular from "angular";
+
+interface StarRatingBindings {
+  numberOfStars?: number;
+  iconSize?: string;
```
Comment (Morkos) [inline app/components/star-rating/app-star-rating.component.ts:5]: number instead? or were you thinking people could pass in "px" or "em"?

File: app/components/star-rating/app-star-rating.component.ts [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+  onChooseRating: ({ rating: number }) => null;
+}
+
+interface IStarRating extends StarRatingBindings {
+  onClickRating: (index: number) => void;
+  onHoverRating: (index: number) => void;
```
Comment (Morkos) [inline app/components/star-rating/app-star-rating.component.ts:11]: onClick?

Comment (Morkos) [inline app/components/star-rating/app-star-rating.component.ts:11]: onHover?

File: app/components/star-rating/app-star-rating.component.ts [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+  onHoverRatingLeave: () => void;
```
Comment (Morkos) [inline app/components/star-rating/app-star-rating.component.ts]: onHoverLeave

File: app/components/star-rating/app-star-rating.component.ts [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+}
+
+class StarRating implements IStarRating {
+  // Bindings
+  numberOfStars?: number;
+  iconSize?: string;
```
Comment (Morkos) [inline app/components/star-rating/app-star-rating.component.ts:18]: number?

**Review by Morkos — APPROVED** [review]
Loving this! Approving this PR. Leaving my comments as optional.

File: app/components/star-rating/app-star-rating.component.ts (hunk 1 above)
Comment (JStein92) [reply app/components/star-rating/app-star-rating.component.ts:5]: yeah I wanted to leave it open

File: app/components/star-rating/app-star-rating.component.ts (hunk 2 above)
Comment (JStein92) [reply app/components/star-rating/app-star-rating.component.ts:11]: I wanted to be more verbose - we are clicking a star rating that has an index, not just clicking the root element.

**Pull request merged**